wee_slack.py

```diff
@@ -2938,20 +2938,51 @@ def msg_command_cb(data, current_buffer, args):
 @slack_buffer_or_ignore
 def command_talk(data, current_buffer, args):
     """
-    Open a chat with the specified user
-    /slack talk [user]
+    Open a chat with the specified user(s)
+    /slack talk [user] ([user2] [user3]...)
     """
 
     data = decode_from_utf8(data)
     args = decode_from_utf8(args)
     e = EVENTROUTER
     team = e.weechat_controller.buffers[current_buffer].team
-    channel_name = args.split(' ')[1]
+    users = args.split(' ')[1:]
     c = team.get_channel_map()
+
+    if len(users) > 1:
+        # Add the current user if not specified since all MPDM names have it
+        if team.nick not in users:
+            users.append(team.nick)
+
+        # Use the user order given if no channels are found below
+        channel_name = 'mpdm-{}-1'.format('--'.join(users))
+
+        # Use the name of any existing channel with the same set of users
+        user_set = set(users)
+        mpdm_regex = re.compile(r'-+')
+        for channel in c:
+            if channel.startswith('mpdm-') and user_set == set(
+                    mpdm_regex.split(channel)[1:-1]):
```

Comment:
As far as I can see, the API doesn't give any guarantees about the name of the mpdm channels, and Slack has lately said that we can't use any usernames as ids, we have to use the actual ids. You should rather compare the `user_ids` tuple you create below with the `members` field of the channels. As well as using `channel.type` instead of startswith mpdm.

Reply:
All great points, thanks! I guess I went with checking the channel name since that's what the DM check was doing and I didn't know about the `members` field.

Should DMs be checking the user ID also then instead of the username? I don't think `SlackDMChannel` is storing the ID currently in `members` but that'd be easy to add.

Reply:
> Should DMs be checking the user ID also then instead of the username?

Yeah, that would be better.

> I don't think SlackDMChannel is storing the ID currently in members but that'd be easy to add.

No, SlackDMChannel doesn't have a `members` field, but it rather has the id in the `user` field. I don't think we should duplicate the information, even though the id isn't likely to change. We could add a `get_members` helper on SlackChannel which just returns `self.member` and override it in SlackDMChannel to return `[self.user]`, if that's clearer.

Reply:
Adding `get_members` simplified the logic so I went with that, thanks for the suggestion!